static_tests/test_not_none.py

```diff
@@ -0,0 +1,12 @@
+from __future__ import annotations
+
+from typing import Any
+from typing_extensions import assert_type
+
+from useful_types import not_none
+
+
+def test(x: Any, y: str | None, z: str) -> None:
+    assert_type(not_none(x), Any)
```

Comment:
Mypy fails on this one:
```
static_tests/test_not_none.py:10: error: Expression is of type <nothing>, not "Any"  [assert-type]
```

Possibly fixable with overloads, let me try that

Reply:
OK, now pyright fails with something worse:

```
  /home/runner/work/useful_types/useful_types/static_tests/test_not_none.py:10:17 - error: "assert_type" mismatch: expected "Any" but received "Never" (reportGeneralTypeIssues)
```

I think I prefer the version without overloads.

Reply:
I think might work with both if you reorder overloads?

Reply:
No, that way it doesn't narrow away the None at all, because None matches the first overload.